lib/vrl/stdlib/src/unnest.rs

```diff
@@ -179,62 +179,40 @@ impl Expression for UnnestFn {
 /// And will remove it returning a set of it's elements.
 ///
 /// For example the typedef for this object:
-/// `{ "nonk" => { "shnoog" => [ { "noog" => 2 }, { "noog" => 3 } ] } }`
+/// `{ "a" => { "b" => [ { "c" => 2 }, { "c" => 3 } ] } }`
 ///
 /// Is converted to a typedef for this array:
-/// `[ { "nonk" => { "shnoog" => { "noog" => 2 } } },
-///    { "nonk" => { "shnoog" => { "noog" => 3 } } },
+/// `[ { "a" => { "b" => { "c" => 2 } } },
+///    { "a" => { "b" => { "c" => 3 } } },
 ///  ]`
 ///
 pub(crate) fn invert_array_at_path(typedef: &TypeDef, path: &LookupBuf) -> TypeDef {
-    use self::value::kind::insert;
+    let kind = typedef.kind().at_path(path);
 
-    let type_def = typedef.at_path(&path.to_lookup());
-
-    let mut array = Kind::from(type_def)
-        .into_array()
-        .unwrap_or_else(Collection::any);
+    let mut array = if let Some(array) = kind.into_array() {
+        array
+    } else {
+        // Guaranteed fallible.
+        // This can't actually be set to "fallible", or it will cause problems due to
```

Comment:
Is this a TODO that we can eventually resolve?

Reply:
Yes, once the linked issue is fixed we should be able to change this.